execute/plugin.go

```diff
@@ -190,7 +216,7 @@ func (p *Plugin) Observation(
 			}
 
 			// TODO: truncate grouped to a maximum observation size?
-			return exectypes.NewObservation(groupedCommits, nil, nil, nil).Encode()
+			return exectypes.NewObservation(groupedCommits, nil, nil, nil, discoveryObs).Encode()
```

Comment:
Suggestion for future PRs: maybe have a special constructor for the observation that is expected to be returned in each state, e.g `NewGetCommitReportsObservation` which would only take two args in this case `groupedCommits` and `discoveryObs` to avoid this `nil, nil, nil` dance which is not super readable. Alternatively we could instantiate the struct directly `Observation{CommitReports: groupedCommits, DiscoveryObs: discoveryObs}`